### PR TITLE
fix(carddav): limit vcard size

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
* Resolves: Card size limit

### CODE REVIEW

**Comment by tcitworld** [comment]
Ah, this reminds me I never added the repair steps I was talking about here: https://github.com/nextcloud/contacts/issues/2743

**Comment by SebastianKrupinski** [comment]
> Ah, this reminds me I never added the repair steps I was talking about here: [nextcloud/contacts#2743](https://github.com/nextcloud/contacts/issues/2743)

@tcitworld Are you going to write the repair step? If you are I modified the PR to use a configurable app setting, please use the same setting for the repair step.

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 1]
```diff
@@ -59,6 +61,7 @@ public function __construct(
 		private IUserManager $userManager,
 		private IEventDispatcher $dispatcher,
 		private Sharing\Backend $sharingBackend,
+		private IConfig $config,
```
Comment (kesselb) [inline apps/dav/lib/CardDAV/CardDavBackend.php:64]: IAppConfig should also work

**Review by hamza221 — APPROVED** [review]

**Review by st3iny — APPROVED** [review]
Looks good to me.

**Pull request merged**